### PR TITLE
Add protoc-gen-connect-es

### PR DESCRIPTION
**Pull request by smaye81**
This adds protoc-gen-connect-es to the plugins repo.

### CODE REVIEW

File: plugins/bufbuild/connect-es/v0.7.0/buf.plugin.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+version: v1
+name: buf.build/bufbuild/connect-es
+plugin_version: v0.7.0
+source_url: https://github.com/bufbuild/connect-es
+description: Generates code that is compatible with browsers and Node.js for working with Connect, gRPC, and gRPC-Web.
+deps:
+  - plugin: buf.build/bufbuild/connect-es:v1.0.0
```
Comment (mfridman) [inline plugins/bufbuild/connect-es/v0.7.0/buf.plugin.yaml:7]: I think the base types should still come from `buf.build/bufbuild/es:v1.0.0` ?

File: plugins/bufbuild/connect-es/v0.7.0/buf.plugin.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+output_languages:
+  - javascript
+  - typescript
+registry:
+  npm:
+    import_style: module
+    rewrite_import_path_suffix: connect.js
+    deps:
+      - package: '@bufbuild/connect-web'
```
[the remark this answers is not in the document]
Comment (smaye81) [reply plugins/bufbuild/connect-es/v0.7.0/buf.plugin.yaml:16]: What does this refer to?  Is it a package on npm?  If so, the package is still named `connect-web` there (for now).

File: plugins/bufbuild/connect-es/v0.7.0/buf.plugin.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+description: Generates code that is compatible with browsers and Node.js for working with Connect, gRPC, and gRPC-Web.
+deps:
+  - plugin: buf.build/bufbuild/connect-es:v1.0.0
+output_languages:
+  - javascript
+  - typescript
+registry:
+  npm:
+    import_style: module
+    rewrite_import_path_suffix: connect.js
```
[the remark this answers is not in the document]
Comment (smaye81) [reply plugins/bufbuild/connect-es/v0.7.0/buf.plugin.yaml:14]: Yep, that is correct.

File: plugins/bufbuild/connect-es/v0.7.0/buf.plugin.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+version: v1
+name: buf.build/bufbuild/connect-es
```
[the remark this answers is not in the document]
Comment (smaye81) [reply plugins/bufbuild/connect-es/v0.7.0/buf.plugin.yaml:2]: Can we just remove `connect-web` from the recommend page until we deprecate it on npm?

File: plugins/bufbuild/connect-es/v0.7.0/buf.plugin.yaml (hunk 1 above)
Comment (smaye81) [reply plugins/bufbuild/connect-es/v0.7.0/buf.plugin.yaml]: Ah good catch.  Thank you.

File: plugins/bufbuild/connect-es/v0.7.0/buf.plugin.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (mfridman) [reply plugins/bufbuild/connect-es/v0.7.0/buf.plugin.yaml:16]: This is the `peerDependencies` for a given remote package, e.g.,

```
"peerDependencies": {
    "@bufbuild/connect-web": "^0.7.0"
}
```

[the remark this answers is not in the document]
Comment (smaye81) [reply plugins/bufbuild/connect-es/v0.7.0/buf.plugin.yaml:16]: Gotcha.  This is good then.

[the remark this answers is not in the document]
Comment (mfridman) [reply plugins/bufbuild/connect-es/v0.7.0/buf.plugin.yaml:16]: I assumed there would be a `@bufbuild/connect-es` library, but I see only https://www.npmjs.com/package/@bufbuild/connect-web is available.

File: plugins/bufbuild/connect-es/v0.7.0/buf.plugin.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (mfridman) [reply plugins/bufbuild/connect-es/v0.7.0/buf.plugin.yaml:2]: Chatted offline.

**Review by mfridman — APPROVED** [review]

**Pull request merged**